sdk/eventhub/eventhubs-checkpointstore-table/src/tableCheckpointStore.ts

```diff
@@ -42,13 +128,77 @@ export class TableCheckpointStore implements CheckpointStore {
    *  - `tracingOptions`: Options for configuring tracing.
    * @returns A list partitions this instance successfully claimed ownership.
    */
-  async claimOwnership(): /*
-    partitionOwnership: PartitionOwnership[],
-    options: OperationOptions = {}
-    */
-  Promise<PartitionOwnership[]> {
-    console.log("nothing to claim");
-    throw new Error("not implemented");
+  async claimOwnership(partitionOwnership: PartitionOwnership[]): Promise<PartitionOwnership[]> {
+    const partitionOwnershipArray: PartitionOwnership[] = [];
```

Comment:
Looks like this comment was automatically closed because the file changed after posting it, so reposting!

So it looks like we have the same bug that python did - we're mutating the list of ownership the user passes in rather than returning completely new ownership items. This is a bug in our storage-blob implementation as well.

We don't want to mutate the ownership that the user passes in, so we should create a copy of the ownership first, then mutate the copy. It's the copy we'll push to the `partitionOwnershipArray`.

We're fortunate that we don't have any nested properties, so to copy an ownership is fairly straightforward, something like this:
```ts
const updatedOwnership = {...ownership};
```

`{...ownership}` creates a new object, and copies every field from `ownership` into the new object. Since we're only dealing with primitive types (strings, numbers) this works fine for our needs.

Reply:
yes this makes sense.